src/libopensc/card-piv.c

```diff
@@ -2965,7 +3213,47 @@ static int piv_pin_cmd(sc_card_t *card, struct sc_pin_cmd_data *data,
 			}
 		}
 	}
-	return iso_drv->ops->pin_cmd(card, data, tries_left);
+
+	priv->pin_cmd_verify_sw1 = 0x00U;
+
+	priv->pin_cmd_verify = 1; /* tell piv_check_sw its a verify to save sw1, sw2 */
+	r = iso_drv->ops->pin_cmd(card, data, tries_left);
+	priv->pin_cmd_verify = 0;
+
+	/* if access to applet is know to be reset by other driver  we select_aid and try again */
+	if ( priv->card_issues & CI_OTHER_AID_LOSE_STATE && priv->pin_cmd_verify_sw1 == 0x6DU) {
+		sc_debug(card->ctx, SC_LOG_DEBUG_NORMAL, "AID may be lost doing piv_find_aid and retry pin_cmd");
```

Comment:
iso_drv->ops->pin_cmd issues the command, then calls sc_check_sw which calls piv_check_sw. 
piv_check_sw  lines 3100-3103 will save priv->pin_cmd_verify_sw1 and sw2  then in line 3120  calls iso_check_sw
and returns to iso_drv->ops->pin_cmd. 
The sw1 and sw2 are saved because pin_cmd does return what may have really happened, and saving the sw1 and sw2 in priv is a way to determine in the piv_pin_cmd may have failed because the PIV applet may have been de-selected, and in line 3226 the opin AID is selected and the pin_cmd is retried in line 3229.